src/schema_cache.rs

```diff
@@ -0,0 +1,41 @@
+use std::sync::OnceLock;
+
+use pyo3::{
+    types::{PyAnyMethods, PyDict},
+    Bound, Py, PyAny, Python,
+};
+
+/// Intended to be a `WeakKeyDictionary[Type, CoreSchema]`
+static CORE_SCHEMA_CACHE: OnceLock<Py<PyAny>> = OnceLock::new();
```

Comment:
is a `WeakKeyDictionary` really the best we can do?

Isn't there a rust data structure we could use here to be a more performance variant of a weakref lookup?

Reply:
I think so yeah, if we do wind up adding a core-side cache for schemas/validators/serializers I expect we could hand write something to avoid dipping into python more than necessary